main.go

```diff
@@ -111,8 +116,15 @@ func run(c *cli.Context) error {
 		go func() { startchan <- struct{}{} }()
 		file := c.String("file")
 		configchan := make(chan bool)
+		// XXX Unwatchable Puppet sources should get events from a timer,
+		// with the same interval as the Puppet agent
+		// (hint: puppet config print runinterval)
 		if !c.Bool("no-watch") {
-			configchan = ConfigWatch(file)
+			if c.IsSet("file") {
+				configchan = ConfigWatch(file)
+			} else if strings.HasSuffix(c.String("puppet"), ".pp") {
+				configchan = ConfigWatch(c.String("puppet"))
```

Comment:
My reasoning is that `mgmt run --puppet /path/to/my.pp` is most akin to `puppet apply` a.k.a. masterless. This is quite different from `puppet agent` in that it *never* uses an interval to re-converge. It has to be triggered by hand (or cron).

Since `mgmt` works differently and will likely *never* launch from cron, I figure watching the manifest is a sound equivalent under our paradigm.